modules/project/shared-vpc.tf

```diff
@@ -17,15 +17,25 @@
 # tfdoc:file:description Shared VPC project-level configuration.
 
 locals {
+  _shared_vpc_agent_config = yamldecode(file("${path.module}/sharedvpc-agent-iam.yaml"))
+  _shared_vpc_agent_config_filtered = [
+    for config in local._shared_vpc_agent_config : config
+    if contains(var.shared_vpc_service_config.grant_iam_for_services, config.service)
```

Comment:
I think you should also filter this with the apis that are enabled in the project. In other words, something like `contains(stintersection(var.services, var.shared_vpc_service_config.grant_iam_for_services), config.service)`

Reply:
we might not have the APIs here Julio

Reply:
We can make it more complex like:
```hcl
    if contains(var.shared_vpc_service_config.services_iam_grants, config.service) && (
      var.services == null || contains(var.services, config.service)
    )
```

If you add more services here, than you activate, the should fail on missing service identity. Filtering here, will make sure that there is no failure at the cost of not letting the user now, that his intent was not fulfilled. 

So I prefer not to add such filtering.